pkg/frontend/querymiddleware/codec.go

```diff
@@ -349,14 +359,21 @@ func (c prometheusCodec) DecodeResponse(ctx context.Context, r *http.Response, _
 	}
 	log.LogFields(otlog.Int("bytes", len(buf)))
 
+	contentType := r.Header.Get("Content-Type")
+	f, ok := knownFormats[contentType]
```

Comment:
Do we need to make sure we're doing some sort of case normalization here because these are headers?

Reply:
No - `Header.Get()` will handle this for us:

> It is case insensitive; textproto.CanonicalMIMEHeaderKey is used to canonicalize the provided key.

([source](https://pkg.go.dev/net/http#Header.Get))